pkg/clusterresourceoverride/internal/handlers/deploy.go

```diff
@@ -197,3 +206,56 @@ func (c *deploymentHandler) EnsureRBAC(context *ReconcileRequestContext, in *aut
 
 	return nil
 }
+
+func DeploymentOverridesMatches(cro *autoscalingv1.ClusterResourceOverride, deployment *appsv1.Deployment) bool {
+	if !replicasMatch(cro, deployment) {
+		return false
+	}
+	if !nodeSelectorMatch(cro, deployment) {
+		return false
+	}
+	if !tolerationsMatch(cro, deployment) {
+		return false
+	}
+	return true
```

Comment:
Oops, I had a comment typed up right here that was supposed to be part of my review, but I must have forgotten to hit the button to add it. I think it was something like:
I like this instead:
```
return replicasMatch(cro, deployment) && nodeSelectorMatch(cro, deployment) &&
    tolerationsMatch(cro, deployment)
```
But I'm fine with it as written if you prefer it.